components/profile.js

```diff
@@ -32,6 +32,51 @@ SteamCommunity.prototype.setupProfile = function(callback) {
 	});
 };
 
+SteamCommunity.prototype.editShowcaseItem = function(showcase, slot, item, callback) {
+	const self = this;
+	//The possible options, with the maximum number of slots and the corresponding type
+	const allowedoptions = {
+		"trade": {
+			"maxslots": 6,
+			"type": 4
+		},
+		"items": {
+			"maxslots": 10,
+			"type": 3
+		},
+		"games": {
+			"maxslots": 4,
+			"type": 2
+		}
+	};
+
+	if(!allowedoptions.hasOwnProperty(showcase)){
+		const err = new Error("The submitted showcase type has no editable items.");
+		return callback ? callback(err) : undefined;
```

Comment:
Using callback here instead of throw is a little counterintuitive for me, but I don't think that's enough to justify change request 🙂.

Reply:
`return undefined;` looks really weird to me. I'd just use an if-statement here.